src/iuse_actor.cpp

```diff
@@ -1428,30 +1436,46 @@ std::unique_ptr<iuse_actor> firestarter_actor::clone() const
     return std::make_unique<firestarter_actor>( *this );
 }
 
-bool firestarter_actor::prep_firestarter_use( const Character &p, map *here, tripoint_bub_ms &pos )
+firestarter_actor::start_type firestarter_actor::prep_firestarter_use( Character &p,
+        map *here, tripoint_bub_ms &pos )
 {
     if( here != &get_map() ) { // Unless 'choose_adjacent' gets map aware.
         debugmsg( "Usage outside reality bubble is not supported" );
-        return false;
+        return start_type::NONE;
     }
 
     // checks for fuel are handled by use and the activity, not here
     if( pos == p.pos_bub( here ) ) {
         if( const std::optional<tripoint_bub_ms> pnt_ = choose_adjacent( _( "Light where?" ) ) ) {
             pos = *pnt_;
         } else {
-            return false;
+            return start_type::NONE;
         }
     }
     if( pos == p.pos_bub( here ) ) {
         p.add_msg_if_player( m_info, _( "You would set yourself on fire." ) );
         p.add_msg_if_player( _( "But you're already smokin' hot." ) );
-        return false;
+        return start_type::NONE;
     }
+
+    const furn_id &f_id = here->furn( pos );
+    const bool is_smoking_rack = f_id == furn_f_metal_smoking_rack ||
+                                 f_id == furn_f_smoking_rack;
+    const bool is_kiln = f_id == furn_f_kiln_empty ||
+                         f_id == furn_f_kiln_metal_empty || f_id == furn_f_kiln_portable_empty;
+
+    if( is_smoking_rack || is_kiln ) {
+        if( is_smoking_rack ) {
+            return iexamine::smoker_prep( p, pos ) ? start_type::SMOKER : start_type::NONE;
+        } else {
+            return iexamine::kiln_prep( p, pos ) ? start_type::SMOKER : start_type::NONE;
```

Comment:
```suggestion
            return iexamine::kiln_prep( p, pos ) ? start_type::KILN : start_type::NONE;
```
i assume?

(and similar to my other comment elsewhere, there's no need for the outer `if (is_smoking_rack || is_kiln)` if you're checking for those in the inner if anyway)

Reply:
Oops, you're entirely right. Those outer if statements are remnants from when there was actually logic being executed before the return is done.